dev/com.ibm.ws.app.manager.wab/src/com/ibm/ws/app/manager/wab/internal/WABInstaller.java

```diff
@@ -568,9 +577,33 @@ protected boolean uninstallFromWebContainer(WAB wab) {
                             wabGroupsLock.unlock();
                         }
                         if (wab.getCreatedApplicationInfo()) {
+                            // system WABs get uninstalled here
                             deployedApp.uninstallApp(wab);
                         } else {
-                            deployedApp.uninstallModule(wab);
+                            final CountDownLatch latch = new CountDownLatch(1);
+                            if (ctx.getBundle(0).getState() == Bundle.STOPPING) {
```

Comment:
should use `ctx.getBundle(org.osgi.framework.Constants.SYSTEM_BUNDLE_LOCATION)` instead of bundle ID `0` here to avoid calling of region framework hooks.